File: c++/functions/capture.cpp
#include <arducam/ArducamCamera.hpp>
#include <cstdlib>
#include <iostream>

void capture(const char* config_path, bool bin_config, int num) {
    Arducam::Camera camera;
    Arducam::Param param;
    param.config_file_name = config_path;  // a path of config file
    param.bin_config = bin_config;         // if the config file is a bin file
    if (!camera.open(param)) {             // open camera, return True if success, otherwise return False
        // get the last error message
        std::cout << "open camera error! " << camera.lastErrorMessage() << "\n";
        std::exit(-1);
    }
    camera.init();  // init camera
    camera.start();
    for (int i = 0; i < num; i++) {
        Arducam::Frame image;
        // the capture return true if success, otherwise return false
        if (camera.capture(image, 1000)) {
            std::cout << "get frame(" << image.format.width << "x" << image.format.height << ") from camera.\n";
        }
    }
    camera.stop();
    camera.close();
}

File: python/function/capture.py
import argparse
from typing import Optional, cast

from ArducamEvkSDK import Camera, Param, get_error_name, Frame


def main(config_path, num):
    """
    @type config_path: str
    @type num: int
    """
    camera = Camera()
    param = Param()
    param.config_file_name = config_path  # a path of config file
    param.bin_config = config_path.endswith(".bin")  # if the config file is a bin file
    if not camera.open(param):  # open camera, return True if success, otherwise return False
        raise Exception("open camera error! {}".format(get_error_name(camera.last_error)))  # get the last error message
    camera.init()  # init camera
    camera.start()
    for i in range(num):
        # the capture should return a Frame object or None
        image = cast(Optional[Frame], camera.capture(1000))
        if image is not None:
            print("get frame({}x{}) from camera.".format(image.format.width, image.format.height))
    camera.stop()
    camera.close()


if __name__ == '__main__':
    parse = argparse.ArgumentParser()
    parse.add_argument(
        "-c", "--config",
        help="Path to config file.",
        type=str,
        required=True,
    )
    parse.add_argument(
        "-n", "--take",
        help="Number of frames to take.",
        type=int,
        default=1,
    )

    args = parse.parse_args()

    main(args.config, args.take)
